### DISCUSSION TITLE
TP7 - Uso de Include Guards y Pragma

### DISCUSSION THREAD

**Discussion by vlzJuan** [INGCOM-UNRN-P1/cursada-2023#186] [Preguntas y respuestas] [answered]
Buenas noches. Hago una consulta respecto a los mismos:

Para evitar doble inclusion de librerias, ¿Podemos usar include guards, tales como

```C
#ifndef SDTIO_H
#define SDTIO_H
#include <stdio.h>
#endif
```

o la directiva de preprocesador 

```C
#pragma once
#include <stdlib.h>
```
?

Siendo sincero, no entiendo mucho el uso de Pragma Once, pero leí que no es soportado por algunos compiladores. En caso de que podamos usarlo, ¿Tiene que usarse antes de cada include? ¿O con usarlo una vez, al principio del archivo, estaria bien?

**Comment by martinvilu** [comment, marked as answer]
No estamos usando headers, para eso necesitamos makefiles que serán explicados en algún momento. (esto forma parte de la _mejor_, o correcta, forma de hacer librerías.
